indico/modules/events/editing/client/js/components/EditableTypeList.jsx

```diff
@@ -0,0 +1,152 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2020 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import editableTypesURL from 'indico-url:event_editing.api_editable_types';
+
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import {Button, Checkbox, Form, Loader} from 'semantic-ui-react';
+import {Form as FinalForm} from 'react-final-form';
+import {Param, Translate} from 'indico/react/i18n';
+import {useIndicoAxios} from 'indico/react/hooks';
+import {FinalCheckbox, FinalSubmitButton, handleSubmitError} from 'indico/react/forms';
+import {indicoAxios} from 'indico/utils/axios';
+import {EditingType} from '../models';
+
+import './EditableTypeList.module.scss';
+
+export default function EditableTypeList({eventId}) {
+  const [editMode, setEditMode] = useState(false);
+  const {data: editableTypes, loading: isLoadingEditableTypes, reFetch} = useIndicoAxios({
+    url: editableTypesURL({confId: eventId}),
+    camelize: true,
+    trigger: eventId,
+  });
+
+  if (isLoadingEditableTypes) {
+    return <Loader inline="centered" active />;
+  } else if (!editableTypes) {
+    return null;
+  }
+
+  const handleSubmit = async formData => {
```

Comment:
Perhaps wrap this inside `useCallback()`. Even though there are no dependencies, every time a re-render happens this will be referentially different.

Reply:
I don't think this is a problem for onSubmit, since that function is only relevant during submission so the function changing should not cause anything to refresh unnecessarily. So using useCallback with a dependency list just adds more complexity to each component using it...

But if you noticed strange behavior or saw something about it in the docs or code of the library, I'd be interested in hearing about it for sure!

Reply:
I agree, the impact shouldn't be relevant, just checked it.
However, we should keep an eye for other cases, because the lack of `useCallback` will cause it to be referentially different and affect the props change within FinalForm whenever a submit is done. Final form uses deep equality. Imo, it's avoidable.

```
// FinalForm.js
function useWhenValueChanges(value, callback, isEqual) {
  if (isEqual === void 0) {
    isEqual = function isEqual(a, b) {
      return a === b;
    };
  }
...
```

Reply:
let's see what the developer of the library @erikras says ;)

https://twitter.com/ThiefMaster/status/1241046930303713282

Reply:
Great idea! 😄 That can be really useful for future cases